website/blog/2022-03-13-2.6.0.md

```diff
@@ -0,0 +1,706 @@
+
+<!-- prettier-ignore -->
+```ts
+// Input
+abstract class Foo {
+  abstract bar(
+    /** comment explaining userId param */
+    userId
+  )
+}
+
+// Prettier 2.5
+abstract class Foo {
+  abstract bar(userId);
+  /** comment explaining userId param */
+}
+
+// Prettier 2.6
+abstract class Foo {
+  abstract bar(
+    /** comment explaining userId param */
+    userId
+  );
+}
+```
+
+#### Fix typecast of superclass in class declarations ([#12222](https://github.com/prettier/prettier/pull/12222) & [#12226](https://github.com/prettier/prettier/pull/12226) by [@duailibe](https://github.com/duailibe))
+
+This was a combination of two separate bugs: moving the comments before the superclass and adding multiple parenthesis around the superclass.
+
+<!-- prettier-ignore -->
+```js
+// Input
+class Foo extends /** @type {Type} */ (Bar) {}
+
+// Prettier 2.5
+class Foo /** @type {Type} */ extends ((Bar)) {}
+
+// Prettier 2.6
+class Foo extends /** @type {Type} */ (Bar) {}
+```
+
+#### Support Unicode Set Notation proposal ([#12241](https://github.com/prettier/prettier/pull/12241) by [@fisker](https://github.com/fisker))
+
+Support [Stage 3 proposal Unicode Set Notation](https://github.com/tc39/proposal-regexp-set-notation/) via [Babel 7.17.0](https://babeljs.io/blog/2022/02/02/7.17.0).
+
+See [the release blog post of Babel v7.17.0](https://babeljs.io/blog/2022/02/02/7.17.0#regexp-set-notation-and-properties-of-strings-14125httpsgithubcombabelbabelpull14125) and [the README of this proposal](https://github.com/tc39/proposal-regexp-set-notation/) for details.
+
+<!-- prettier-ignore -->
+```jsx
+// Examples
+
+/[\p{Decimal_Number}--[0-9]]/v; // Non-ASCII decimal digits
+
+"Did you see the 👩🏿‍❤️‍💋‍👩🏾 emoji?".match(/\p{RGI_Emoji}/v). // ["👩🏿‍❤️‍💋‍👩🏾"]
+
+/[\r\n\q{\r\n|NEWLINE}]/v; // Matches \r, \n, \r\n or NEWLINE
+```
+
+#### Add parens to `ClassExpression` with decorators ([#12260](https://github.com/prettier/prettier/pull/12260) by [@fisker](https://github.com/fisker))
+
+<!-- prettier-ignore -->
+```js
+// Input
+(@f() class {});
+
+// Prettier 2.5
+@f()
+class {};
+
+// Prettier 2.5 (Second format)
+SyntaxError: A class name is required. (2:7)
+  1 | @f()
+> 2 | class {};
+    |       ^
+  3 |
+
+// Prettier 2.6
+(
+  @f()
+  class {}
+);
+```
+
+#### Improve printing of comments in type aliases in Flow & TS ([#12268](https://github.com/prettier/prettier/pull/12268) by [@duailibe](https://github.com/duailibe))
+
+For Flow, the comments will now be more aligned to how we print comments in assignments where the right-hand side is an object expression:
+
+<!-- prettier-ignore -->
+```js
+// Input
+type Props = // comment explaining the props
+  {
+    isPlaying: boolean
+  };
+
+// Prettier 2.5
+// comment explaining the props
+type Props = {
+  isPlaying: boolean,
+};
+
+// Prettier 2.6
+type Props =
+  // comment explaining the props
+  {
+    isPlaying: boolean,
+  };
+```
+
+And for TS that comment would not be stable with a second format:
```

Comment:
```suggestion
This change also fixes an issue where a similarly-placed comment in TypeScript would be moved again after a second formatting operation:
```

(still not 100% happy with this wording but I think it might be clearer)